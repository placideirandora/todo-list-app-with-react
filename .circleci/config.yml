version: 2.1
jobs:
  build:
    docker:
      - image: circleci/node:10.16.0

    working_directory: ~/todo-list-app-with-react-js

    steps:
      - checkout
      - restore_cache:
          keys:
            - v1-dependencies-{{ checksum "package.json" }}
            - v1-dependencies-

      - run: npm install

      - save_cache:
          paths:
            - node_modules
          key: v1-dependencies-{{ checksum "package.json" }}

      - run: npm run test:coverage
