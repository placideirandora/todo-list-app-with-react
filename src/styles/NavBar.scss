@import 'Colors';
@import 'Fonts';

nav {
  background: $lightBlue;
  .logo {
    margin-left: 3rem;
    font-family: $saira;
    font-size: 4rem;
    color: $white;
    text-decoration: none;
  }
  .menu {
    float: right;
    font-size: 2rem;
    color: $white;
    text-decoration: none;
    padding: 5px;
    font-family: $saira;
    line-height: 6rem;
  }

  .second {
    margin-left: 1.8rem;
  }

  .last {
    margin-right: 3rem;
    margin-left: 1.8rem;
  }
}
