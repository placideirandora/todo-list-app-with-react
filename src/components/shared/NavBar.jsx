import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import { navBarStyles } from '../../styles/navBar';

const Navbar = () => {
  return (
    <Fragment>
      <nav style={navBarStyles.navBar}>
        <Link to="/" style={navBarStyles.logo}>
          TO-DO LIST
        </Link>
        <menu style={navBarStyles.menuWrapper}>
          <Link to="/Todo" style={navBarStyles.menu}>
            VIEW TODOS
          </Link>
          <Link to="/AddTodo" style={navBarStyles.menu}>
            ADD TODOS
          </Link>
          <Link to="/About" style={navBarStyles.menu}>
            ABOUT
          </Link>
        </menu>
      </nav>
    </Fragment>
  );
};

export default Navbar;
