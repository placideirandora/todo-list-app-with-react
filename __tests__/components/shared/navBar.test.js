import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as adapter from '../../config/enzyme.config';
import NavBar from '../../../src/components/shared/NavBar';

describe('NavBar Component Test Cases', () => {
  it('should render the component without any failure', () => {
    const component = shallow(<NavBar />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
