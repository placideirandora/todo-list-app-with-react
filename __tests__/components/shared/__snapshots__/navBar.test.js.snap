// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component Test Cases should render the component without any failure 1`] = `
<Fragment>
  <nav
    style={
      Object {
        "backgroundColor": "#46a1de",
        "fontFamily": "Saira Stencil One",
        "padding": "0 10%",
      }
    }
  >
    <Link
      style={
        Object {
          "color": "white",
          "fontSize": "3.5rem",
          "textDecoration": "none",
        }
      }
      to="/"
    >
      TO-DO LIST
    </Link>
    <menu
      style={
        Object {
          "float": "right",
          "marginTop": "1.5rem",
        }
      }
    >
      <Link
        style={
          Object {
            "color": "white",
            "fontSize": "2rem",
            "padding": "1rem",
            "textDecoration": "none",
          }
        }
        to="/Todo"
      >
        VIEW TODOS
      </Link>
      <Link
        style={
          Object {
            "color": "white",
            "fontSize": "2rem",
            "padding": "1rem",
            "textDecoration": "none",
          }
        }
        to="/AddTodo"
      >
        ADD TODOS
      </Link>
      <Link
        style={
          Object {
            "color": "white",
            "fontSize": "2rem",
            "padding": "1rem",
            "textDecoration": "none",
          }
        }
        to="/About"
      >
        ABOUT
      </Link>
    </menu>
  </nav>
</Fragment>
`;
